### PR TITLE
Add changes related to API version 2018-08-23

### PR DESCRIPTION
**Pull request by remi-stripe**
This adds changes related to the latest API version and new properties introduced
* add support for `tax_info` and `tax_info_verification` on Customer
* add support for `unit_amount` on Tier and deprecate `amount`

r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/model/Plan.java [hunk 1]
```diff
@@ -193,8 +193,18 @@ public Plan update(Map<String, Object> params, RequestOptions options) throws St
...
     Long upTo;
+
+    /**
+     * The {@code amount} attribute.
+     *
+     * @return the {@code amount} attribute
+     * @deprecated Prefer using the {@code unitAmount} attribute instead.
+     * @see <a href="https://stripe.com/docs/upgrades#2018-08-23">API version 2018-08-23</a>
+     */
+    @Deprecated
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/Plan.java]: Is this attribute really deprecated? It's still present in the API reference: https://stripe.com/docs/api#plan_object-amount

Comment (remi-stripe) [reply src/main/java/com/stripe/model/Plan.java]: It is actively deprecated since we shipped `unit_amount`. We're renaming amount because we are going to ship another type of amount property and we wanted to to avoid the confusion.

TL;DR: yep, the docs are incorrect I suppose

**Comment by ob-stripe** [comment]
Can you flag the `businessVatId` on `Customer` as `@Deprecated` in favor of `taxInfo`?

**Comment by remi-stripe** [comment]
@ob-stripe done PTAL

File: src/main/java/com/stripe/model/Customer.java [hunk 2]
```diff
@@ -33,6 +32,17 @@
...
+  TaxInfoVerification taxInfoVerification;
+
+  /**
+   * The {@code businessVatId} attribute.
+   *
+   * @return the {@code businessVatId} attribute
+   * @deprecated Prefer using the {@link #taxInfo} attribute instead.
+   * @see <a href="https://stripe.com/docs/upgrades#2018-08-23">API version 2018-08-23</a>
+   */
+  String businessVatId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/Customer.java]: You need to annotate the attribute itself with `@Deprecated`.

**Comment by remi-stripe** [comment]
@ob-stripe updated, sorry for the back and forth. PTAL

**Comment by ob-stripe** [comment]
Released as 6.8.0.
